ui/raidboss/data/06-ew/ultimate/dragonsongs_reprise_ultimate.ts

```diff
@@ -229,6 +251,218 @@ const triggerSet: TriggerSet<Data> = {
+      netRegexKo: NetRegexes.ability({ id: '63D3', source: '기사신 토르당', capture: false }),
+      condition: (data) => data.phase === 'thordan',
+      delaySeconds: 4.5,
+      promise: async (data) => {
+        // Collect Ser Vellguine (3636), Ser Paulecrain (3637), Ser Ignasse (3638) entities
+        const vellguineLocaleNames: LocaleText = {
+          en: 'Ser Vellguine',
+        };
+
+        const paulecrainLocaleNames: LocaleText = {
+          en: 'Ser Paulecrain',
+        };
+
+        const ignasseLocaleNames: LocaleText = {
+          en: 'Ser Ignasse',
+        };
+
+        // Select the knights
+        const combatantNameKnights = [];
+        combatantNameKnights.push(vellguineLocaleNames[data.parserLang]);
+        combatantNameKnights.push(paulecrainLocaleNames[data.parserLang]);
+        combatantNameKnights.push(ignasseLocaleNames[data.parserLang]);
+
+        const spiralThrusts = [];
+        for (const combatantName of combatantNameKnights) {
+          let combatantData = null;
+          if (combatantName) {
+            combatantData = await callOverlayHandler({
+              call: 'getCombatants',
+              names: [combatantName],
+            });
+          }
+
+          // if we could not retrieve combatant data, the
+          // trigger will not work, so just resume promise here
+          if (combatantData === null) {
+            console.error(`Spiral Thrust: null data`);
+            return;
+          }
+          if (!combatantData.combatants) {
+            console.error(`Spiral Thrust: null combatants`);
+            return;
+          }
+          const combatantDataLength = combatantData.combatants.length;
+          if (combatantDataLength !== 1) {
+            console.error(`Spiral Thrust: expected 1 combatants got ${combatantDataLength}`);
+            return;
+          }
+
+          // Add the combatant's position
+          const combatant = combatantData.combatants.pop();
+          if (!combatant)
+            throw new UnreachableCode();
+          spiralThrusts.push(matchedPositionToDir(combatant));
+        }
+
+        if (spiralThrusts.length !== 3) {
+          console.error(`Spiral Thrusts: expected 3 combatants got ${spiralThrusts.length}`);
+          return;
+        }
+
+        // Array of dirNums
+        const dirNums = [0, 1, 2, 3, 4, 5, 6, 7];
+
+        // Remove where the knights are at and where they will go to
+        if (spiralThrusts[0] || spiralThrusts[0] === 0) {
+          delete dirNums[(spiralThrusts[0] + 4) % 8];
+          delete dirNums[spiralThrusts[0]];
+        }
+        if (spiralThrusts[1] || spiralThrusts[1] === 0) {
+          delete dirNums[(spiralThrusts[1] + 4) % 8];
+          delete dirNums[spiralThrusts[1]];
+        }
+        if (spiralThrusts[2] || spiralThrusts[2] === 0) {
+          delete dirNums[(spiralThrusts[2] + 4) % 8];
+          delete dirNums[spiralThrusts[2]];
+        }
```

Comment:
What are these conditionals trying to do? If you're just trying to satisfy typescript, maybe this instead? (Please give this suggestion a whirl in raid emulator or something, as I'm just typing it, but I think it's just a little odd to say `if (spiralThrusts[0] || spiralThrusts[0] === 0) {`

```suggestion
        const [thrust0, thrust1, thrust2] = spiralThrusts;
        if (thrust0 === undefined || thrust1 === undefined || thrust2 === undefined)
          return;

        // Array of dirNums
        const dirNums = [0, 1, 2, 3, 4, 5, 6, 7];

        // Remove where the knights are at and where they will go to
        delete dirNums[(thrust0 + 4) % 8];
        delete dirNums[thrust0];
        delete dirNums[(thrust1 + 4) % 8];
        delete dirNums[thrust1];
        delete dirNums[(thrust2 + 4) % 8];
        delete dirNums[thrust2];
```

Reply:
Problem I was having is I was pulling from an index of the spiralThrust array, which could potentially be null and so I had added a simple if spiralThrusts[0] as a check for it. However an oversight in that the integer value of 0 evaluates to false and so I added an exception. At the time I couldn't see a cleaner fix for it, was making modifications between pulls haha.

Reply:
Haha, I know that feeling.  :)